Logger added. Small sugar fixes.

wjdda opened the pull request:
1) Log4j Logger.
 ++ gamelogger/LogConfFactory. Configurating root logger (for every new
 loggers in classes).
 ++ gamelogger/LogOptions. Constants to simple configurating loggers.

 2) Other classes:
 <> printMsg -> LOG.
 <> Some small code corrections.
 ++ Added comments "QUESTION", "TEST" (For IDE marker parser).

 3) Main.
 -- printMsg().
 ++ initLogger(). Configure root logger for future children generation.
 ++ initControllers(). Logger in MouseController creating earlier than
 root logger. Moved initialization here.

wjdda committed:
Logger added. Small sugar fixes.

 1) Log4j Logger.
 ++ gamelogger/LogConfFactory. Configurating root logger (for every new
 loggers in classes).
 ++ gamelogger/LogOptions. Constants to simple configurating loggers.

 2) Other classes:
 <> printMsg -> LOG.
 <> Some small code corrections.
 ++ Added comments "QUESTION", "TEST" (For IDE marker parser).

 3) Main.
 -- printMsg().
 ++ initLogger(). Configure root logger for future children generation.
 ++ initControllers(). Logger in MouseController creating earlier than
 root logger. Moved initialization here.

wjdda commented:
Don't forget project in Trello.

wjdda committed:
Log fixes by GitHub remarks.

1) Artefacts: "getx", "gety" - in commented logs (GameMap, D_Thread).
2) English messages corrected (Main, LogConfFactory).
3) Added to Layouts %tid. (LogOptions).
4) Corrected expression for Main.debugMargins().

...and etc.

AlexanderSamoilov commented on src/com/company/gamelogger/LogConfFactory.java:
```diff
@@ -0,0 +1,184 @@
+        builder.setStatusLevel(LEVEL);
+
+        String logFilePath = FILE_PATH + "/" + FILE_NAME;
+        boolean debugging = LEVEL.equals(Level.DEBUG) || LEVEL.equals(Level.ALL);
```
This is more correct:
boolean debugging = LEVEL.getlevel() <= Level.DEBUG;

wjdda replied:
See new commit!

AlexanderSamoilov commented on src/com/company/gamelogger/LogOptions.java:
```diff
@@ -0,0 +1,42 @@
+
+    // Example of good format of output:
+    //    2019-04-16 22:06:54 [22] gamethread.Main > mainfunc() INFO: The game starts.
+    static final String STDOUT_LAYOUT = "%d{yyyy-MM-dd HH:mm:ss} [%-2.2tid] %-30.30c{2} > %20.20M() %5level: %msg%n%throwable";
```
I propose to use just %tid. You never know what numbers can Java threads have. It is very OS-dependent.
I saw many times on Linux the output of "jmap" and it showed me bigger numbers.
The longest which I saw was always 5 digits which makes me think that the maximal thread ID is _probably_ 32767, however I am not sure.
That is why I propose to not predict it at let your OS to write the comlpete thread ID without making it cut.

wjdda replied:
Added 10 digits to tid. If we don't do this, the layout breaks down.

AlexanderSamoilov commented on src/com/company/gamecontent/GameMap.java:
```diff
@@ -374,9 +391,9 @@ public boolean occupied(Rectangle givenRect, GameObject exceptObject) {
-                    //Main.printMsg("Check 1: (" + thisObjRect.x + "," + thisObjRect.y + "," + thisObjRect.width + "," + thisObjRect.height);
-                    //Main.printMsg("Check 2: (" + objOnBlockRect.getX() + "," + objOnBlockRect.getY() + "," + objOnBlockRect.getWidth() + "," + objOnBlockRect.getHeight());
-                    //Main.printMsg("Check 3: (" + objOnBlock.getAbsLoc()[0] + "," + objOnBlock.getAbsLoc()[1] + "," + objOnBlock.getAbsSize()[0] + "," + objOnBlock.getAbsSize()[1]);
+//                    LOG.debug("Check 1: (" + thisObjRect.x + "," + thisObjRect.y + "," + thisObjRect.width + "," + thisObjRect.height);
```
I see you forgot to answer my comment about it.
I asked you why did you change function calls inside these 3 commented lines. I suppose that you wrongly use the updated API from your local workspace and did not revert it before doing commit.

wjdda replied:
Now I have done a log version 2.x import everywhere.

wjdda committed:
Second fixes by remarks from GitHub.

1) For All Classes. Imports "org.apache.log4j..." was not correct, because it's from 1.x ver. Fixed it to 2.x
version: "org.apache.logging.log4j...".
2) Corrected levels checks in LogConfFactory and Main.debugMargins.
3) Finaly fixed comment in GameMap.
